src/NetMQ/NetMQPoller.cs

```diff
@@ -330,7 +387,7 @@ public void Remove([NotNull] NetMQTimer timer)
 
             timer.When = -1;
 
-            Run(() => m_timers.Remove(timer));
+            RunAsync(() => m_timers.Remove(timer));
```

Comment:
Is there a reason you didn't create `RemoveAsync(NetMQTimer)`?

Reply:
I was just focused on the sockets, but that should follow the same pattern.  Once we get the socket-related updates to everyone's liking, we should be able to copy for  `NetMQTimer`?

Reply:
added overload and tesst